MSW: mocks reference non existing function for empty examples

### Description

When an OpenAPI spec includes a path with no example, the MSW mocks generate a reference to a function that does not exist. This is a regression caused around 8.4.0 - this did not happen before then.


### Output client

axios

### Configuration (orval.config)

```ts
import { camelCase, capitalize } from 'lodash-es';
import { defineConfig } from 'orval';

export default defineConfig({
  'user-management': {
    output: {
      workspace: 'src/',
      indexFiles: true,
      mode: 'tags-split',
      target: './endpoints',
      schemas: './model',
      client: 'react-query',
      httpClient: 'axios',
      baseUrl: '/users',
      headers: true,
      urlEncodeParameters: true,
      allParamsOptional: true,
      mock: {
        type: 'msw',
        delay: false,
        useExamples: true,
        baseUrl: '/users',
      },
      override: {
        useNamedParameters: true,
        useTypeOverInterfaces: true,
        mutator: {
          path: './fetch.ts',
          name: 'fetch',
        },
        query: {
          useQuery: true,
          useMutation: true,
          useSuspenseQuery: true,
          shouldSplitQueryKey: true,
        },
      },
    },
    input: {
      target: './openapi.json',
      filters: {
        mode: 'exclude',
        tags: [/health/, /schema/],
      },
    },
  },
});

```


### Environment

```
  System:
    OS: macOS 26.3.1
    CPU: (14) arm64 Apple M4 Pro
    Memory: 175.88 MB / 24.00 GB
    Shell: 5.9 - /bin/zsh
  npmPackages:
    @tanstack/react-query: 5.69.0 => 5.69.0 
    axios: ^1.15.0 => 1.15.0 
    msw: ^2.12.7 => 2.12.7 
    orval: 8.2.0 => 8.2.0 
    react: 18.3.1 => 18.3.1 
    zod: ^3.25.69 => 3.25.69 
```

### Expected behavior

Orval 8.2.0

```
export const getGetApiV1UsersUserHashConfirmNewEmailMockHandler = (overrideResponse?: unknown | ((info: Parameters<Parameters<typeof http.get>[1]>[0]) => Promise<unknown> | unknown), options?: RequestHandlerOptions) => {
  return http.get('/users/api/v1/users/:userHash/confirm_new_email', async (info) => {
  if (typeof overrideResponse === 'function') {await overrideResponse(info); }
    return new HttpResponse(null,
      { status: 200,
        
      })
  }, options)
}
```

### Actual behavior

Orval 8.8.1

```
export const getGetApiV1UsersUserHashConfirmNewEmailMockHandler = (overrideResponse?: unknown | ((info: Parameters<Parameters<typeof http.get>[1]>[0]) => Promise<unknown> | unknown), options?: RequestHandlerOptions) => {
  return http.get('/users/api/v1/users/:userHash/confirm_new_email', async (info: Parameters<Parameters<typeof http.get>[1]>[0]) => {
  if (typeof overrideResponse === 'function') {await overrideResponse(info); }
  const resolvedBody = overrideResponse !== undefined
    ? (typeof overrideResponse === "function" ? await overrideResponse(info) : overrideResponse)
    : getGetApiV1UsersUserHashConfirmNewEmailResponseMock();
    const textBody = typeof resolvedBody === 'string' ? resolvedBody : JSON.stringify(resolvedBody ?? null);
    return new HttpResponse(null,
      { status: 200
      })
  }, options)
}
```

Typescript checks:
```
queries/user-management/src/endpoints/api-v1-user/api-v1-user.msw.ts:73:7 - error TS2552: Cannot find name 'getGetApiV1UsersUserHashConfirmNewEmailResponseMock'. Did you mean 'getPutApiV1UsersUserHashConfirmResponseMock'?

73     : getGetApiV1UsersUserHashConfirmNewEmailResponseMock();
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### OpenAPI document (minimal, if applicable)

```yaml
openapi: 3.0.3
info:
  title: Demo
  version: 0.0.0
paths: {
  "/api/v1/users/{user_hash}/confirm_new_email": {
      "get": {
        "summary": "confirm_new_email",
        "tags": [
          "Api::V1::User"
        ],
        "parameters": [
          {
            "name": "user_hash",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            },
            "example": "HASH-WHATTAHASH-21"
          }
        ],
        "responses": {
          "302": {
            "description": "does not sign out the user",
            "content": {
              "text/html": {
                "schema": {
                  "type": "string"
                }
              }
            }
          }
        }
      }
    }
}
```


### Additional context

_No response_

--- a/packages/mock/src/msw/index.test.ts
+++ b/packages/mock/src/msw/index.test.ts
@@ -384,6 +384,31 @@ describe('generateMSW', () => {
       expect(result.implementation.handler).not.toContain('HttpResponse.text(');
     });
 
+    // Regression test for https://github.com/orval-labs/orval/issues/3270.
+    // When the response definition resolves to `unknown` (as with a 302
+    // redirect returning text/html that is not treated as a success body),
+    // no `*ResponseMock` helper is generated. The handler must not emit a
+    // `resolvedBody`/`textBody` prelude that references the missing helper.
+    it('should not reference a missing *ResponseMock for unknown text responses', () => {
+      const unknownHtmlVerbOptions = {
+        ...mockVerbOptions,
+        response: {
+          ...mockVerbOptions.response,
+          definition: { success: 'unknown' },
+          types: { success: [{ key: '302', value: 'unknown' }] },
+          contentTypes: ['text/html'],
+        },
+      } as GeneratorVerbOptions;
+
+      const result = generateMSW(unknownHtmlVerbOptions, baseOptions);
+
+      expect(result.implementation.function).toBe('');
+      expect(result.implementation.handler).not.toContain('ResponseMock()');
+      expect(result.implementation.handler).not.toContain('resolvedBody');
+      expect(result.implementation.handler).not.toContain('textBody');
+      expect(result.implementation.handler).toContain('new HttpResponse(null,');
+    });
+
     it('should use HttpResponse.xml for vendor +xml responses', () => {
       const vendorXmlVerbOptions = {
         ...mockVerbOptions,

--- a/packages/mock/src/msw/index.ts
+++ b/packages/mock/src/msw/index.ts
@@ -250,15 +250,19 @@ function generateDefinition(
 
   let responseBody: string;
   // Use a prelude to evaluate the override expression once into a temp variable
-  // (the expression contains `await` so must not be duplicated)
+  // (the expression contains `await` so must not be duplicated). Only emit it
+  // when we actually generate a `*ResponseMock()` helper — otherwise the
+  // prelude would reference a function that doesn't exist (issue #3270).
   let responsePrelude = '';
-  if (isBinaryResponse) {
-    responsePrelude = `const binaryBody = ${resolvedResponseExpr};`;
-  } else if (isVoidUnionType || needsRuntimeContentTypeSwitch) {
-    responsePrelude = `const resolvedBody = ${resolvedResponseExpr};`;
-  } else if (isTextResponse && !shouldPreferJsonResponse) {
-    responsePrelude = `const resolvedBody = ${resolvedResponseExpr};
+  if (isReturnHttpResponse) {
+    if (isBinaryResponse) {
+      responsePrelude = `const binaryBody = ${resolvedResponseExpr};`;
+    } else if (isVoidUnionType || needsRuntimeContentTypeSwitch) {
+      responsePrelude = `const resolvedBody = ${resolvedResponseExpr};`;
+    } else if (isTextResponse && !shouldPreferJsonResponse) {
+      responsePrelude = `const resolvedBody = ${resolvedResponseExpr};
     const textBody = typeof resolvedBody === 'string' ? resolvedBody : JSON.stringify(resolvedBody ?? null);`;
+    }
   }
   if (!isReturnHttpResponse) {
     responseBody = `new HttpResponse(null,

